juju/model.py

```diff
@@ -114,31 +127,39 @@ def __getitem__(self, relation_name):
 
 class Relation:
     def __init__(self, relation_name, relation_id, local_unit, backend, cache):
-        self.relation_name = relation_name
-        self.relation_id = relation_id
-        self._apps = set()
-        self._units = []
-        for unit_name in backend.relation_list(relation_id):
+        self.name = relation_name
+        self.id = int(relation_id.split(':')[-1]) if isinstance(relation_id, str) else relation_id
```

Comment:
Per prior conversations, we shouldn't accept bogus data inside our model. This kind of transformation should be done "at the door", wherever the non-conforming values are found.